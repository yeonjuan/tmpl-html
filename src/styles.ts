import {hasOwn} from './utils';

type Value = string | number | boolean | undefined | null;
type RecordValue = Record<string, Value>;

export default function styles(...args: RecordValue[]): string {
  const styleMap: any = {};
  const style: string[] = [];

  for (let i = 0; i < args.length; i++) {
    const arg = args[i];
    if (typeof arg === 'object' && !!arg) {
      for (const key in arg) {
        if (arg[key] && hasOwn.call(arg, key)) {
          styleMap[key] = arg[key];
        }
      }
    }
  }

  for (const key in styleMap) {
    if (key && styleMap[key] && hasOwn.call(styleMap, key)) {
      style.push(`${key}:${styleMap[key]}`);
    }
  }

  return `"${style.join(';')}"`;
}
