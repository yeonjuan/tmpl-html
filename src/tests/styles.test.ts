import styles from '../styles';

test.each([
  [{width: '100px'}],
  [{width: '100px', height: '50px'}],
  [{width: null, height: '50px'}],
  [{width: '100px', height: undefined}],
])('styles', (value) => {
  expect(styles(value)).toMatchSnapshot()
});
