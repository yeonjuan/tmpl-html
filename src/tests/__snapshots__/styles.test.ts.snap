// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles 1`] = `"\\"width:100px\\""`;

exports[`styles 2`] = `"\\"width:100px;height:50px\\""`;

exports[`styles 3`] = `"\\"height:50px\\""`;

exports[`styles 4`] = `"\\"width:100px\\""`;
